be/src/item/item.service.ts

```diff
@@ -22,8 +22,10 @@ export class ItemService {
     return this.itemRepository.find({});
   }
 
-  findOne(id: number) {
-    return this.itemRepository.findOne(id);
+  async findOne(id: number) {
+    const item = await this.itemRepository.findOne(id);
+    if (item) return this.itemRepository.findOne(id);
```

Comment:
I think it makes better sense for the logic to be the other way but this will suffice